ui/webui/src/components/installation/InstallationLanguage.jsx

```diff
@@ -82,31 +84,34 @@ class LanguageSelector extends React.Component {
             this.setState({ selectedItem: lang });
         };
 
-        const isLoading = languages.length !== locales.length;
+        const isLoading = languages.length === 0 || languages.length !== locales.length;
+
+        if (isLoading) {
+            return <EmptyStatePanel loading />;
+        }
+
         const options = (
-            !isLoading
-                ? locales.map(langLocales => {
-                    const currentLang = languages.find(lang => getLanguageId(lang) === getLanguageId(langLocales[0]));
-
-                    return (
-                        <SelectGroup
-                          label={cockpit.format("$0 ($1)", getLanguageNativeName(currentLang), getLanguageEnglishName(currentLang))}
-                          key={getLanguageId(currentLang)}>
-                            {langLocales.map(locale => (
-                                <SelectOption
-                                  id={getLocaleId(locale).split(".UTF-8")[0]}
-                                  key={getLocaleId(locale)}
-                                  value={{
-                                      toString: () => getLocaleNativeName(locale),
-                                      // Add a compareTo for custom filtering - filter also by english name
-                                      localeId: getLocaleId(locale)
-                                  }}
-                                />
-                            ))}
-                        </SelectGroup>
-                    );
-                })
-                : []
+            locales.map(langLocales => {
+                const currentLang = languages.find(lang => getLanguageId(lang) === getLanguageId(langLocales[0]));
+
+                return (
+                    <SelectGroup
+                      label={cockpit.format("$0 ($1)", getLanguageNativeName(currentLang), getLanguageEnglishName(currentLang))}
```

Comment:
The `cockpit.format()` function looks interesting! :) 

https://www.malasuk.com/doc/cockpit/cockpit-util.html 
```
Format a string interpolating args into template using shell like syntax. The args may be either an array or javascript object. The template can contain fields that look like $name or ${name} or $0. Numeric fields are used with array args and start at zero.

In the second form, multiple arg arguments may be passed directly, and interpolated as as numeric fields in the template.

All falsy arguments except the numbers 0 and 0.0are replaced by an empty string.
```